### PR TITLE
vsc-modification

### PR DESCRIPTION
**Pull request by jamal-khey**

### CODE REVIEW

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-pane.tsx [hunk 1]
```diff
@@ -74,7 +81,20 @@ const ConverterStationPane: FunctionComponent<VscConverterStationPaneProps> = ({
         }
     }, [studyUuid, currentNodeUuid]);
 
-    const generatorIdField = (
+    const previousDataany = previousValues as any;
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/converter-station/converter-station-pane.tsx:84]: Is the "any" cast is necessary ?

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx [hunk 2]
```diff
@@ -176,6 +232,69 @@ export function getConverterStationCreationData(converterStation: any) {
...
+        // ),
+        // connectionDirection:
+        //     toModificationOperation(converterStation[CONNECTIVITY]?.[CONNECTION_DIRECTION] ??
+        //     UNDEFINED_CONNECTION_DIRECTION),
+        // connectionPosition: toModificationOperation(
+        //     converterStation[CONNECTIVITY]?.[CONNECTION_POSITION]
+        // ),
+        // connected: toModificationOperation(
+        //     converterStation[CONNECTIVITY]?.[CONNECTED]
+        // ),
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx:278]: maybe remove the comments if the code is not used anymore

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                )}
+            </ModificationDialog>
+        </FormProvider>
+    );
+};
+
+export default VscModificationDialog;
+function snackError(arg0: { messageTxt: any; headerId: string }) {
+    throw new Error('Function not implemented.\n' + arg0.messageTxt);
+}
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx:274]: I would prefer this function to be above the VscModificationDialog one

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx [hunk 4]
```diff
@@ -234,13 +238,16 @@ export function getConverterStationCreationData(converterStation: any) {
...
+        | undefined
 ) {
-    const reactiveLimits = converterStation[REACTIVE_LIMITS];
+    const reactiveLimits = converterStation[REACTIVE_LIMITS];//should return 3 //OK
     const buildCurvePointsToStore = calculateCurvePointsToStore(
         reactiveLimits[REACTIVE_CAPABILITY_CURVE_TABLE],
-        converterStation
+        converterStationToModify
     );
+    console.log("🚀-debug ~ buildCurvePointsToStore:", buildCurvePointsToStore);
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx:250]: Don't forget to remove dev logs

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-pane.tsx (hunk 1 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/converter-station/converter-station-pane.tsx]: removed

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx [hunk 5]
```diff
@@ -124,14 +154,40 @@ export function getVscConverterStationSchema(id: string) {
...
+            [REACTIVE_POWER]: yup.number().nullable(),
+            // .when([VOLTAGE_REGULATION_ON], {
+            //     is: false,
+            //     then: (schema) => schema.required(),
+            // }),
+            [VOLTAGE]: yup.number().nullable(),
+            // .when([VOLTAGE_REGULATION_ON], {
+            //     is: true,
+            //     then: (schema) => schema.required(),
+            // }),
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx]: removed

File: src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx (hunk 4 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/converter-station/converter-station-utils.tsx]: removed

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx (hunk 3 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx:388]: done

File: src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx [hunk 6]
```diff
@@ -42,6 +42,7 @@ const VscCreationForm: FunctionComponent<VscCreationFormProps> = ({
                     currentNode={currentNode}
                     id={CONVERTER_STATION_1}
                     stationLabel={'converterStation1'}
+                    isModification={false}
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx:45]: you may not write it, and use isModification for true value only (I'm not sure if you should change type isModification? in the ConverterStationPaneProps)

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            <Box
+                hidden={tabIndex !== VSC_CREATION_TABS.CONVERTER_STATION_1}
+                p={1}
+            >
+                <ConverterStationPane
+                    studyUuid={studyUuid}
+                    currentNode={currentNode}
+                    id={CONVERTER_STATION_1}
+                    stationLabel={'converterStation1'}
+                    isModification={true}
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx:120]: you can change it for isModification

File: src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx (hunk 6 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx:45]: do you mean make isModification = false by default ? so we don't need to specify it explicitly ?

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx [hunk 8]
```diff
@@ -212,6 +264,63 @@ const VscModificationDialog: React.FC<any> = ({
...
+
+    const updatePreviousReactiveCapabilityCurveTableConverterStation1 = (
+        action: string,
+        index: number
+    ) => {
+        setVcsToModify((previousValue: VscModificationInfo | null) => {
+            const newRccValues =
+                previousValue?.converterStation1?.reactiveCapabilityCurveTable;
+            console.log('debug', 'previousValue', previousValue);
+            console.log('debug', 'newRccValues', newRccValues);
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx]: you may remove the logs

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx [hunk 9]
```diff
@@ -223,7 +332,7 @@ const VscModificationDialog: React.FC<any> = ({
             <ModificationDialog
                 fullWidth
                 onClear={setValuesAndEmptyOthers}
-                onSave={onSubmit}
+                onSave={handleSubmit(onSubmit)}
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx:342]: don't you have to use a callback like () => handleSubmit(onSubmit)

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx (hunk 7 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx:120]: i don't undestand, do you mean make a variable ?

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx [hunk 10]
```diff
@@ -32,6 +33,12 @@ interface VscModificationFormProps {
     vscInfos: VscModificationInfo | null;
     setTabIndex: React.Dispatch<React.SetStateAction<number>>;
     tabIndexesWithError: number[];
+    updatePreviousReactiveCapabilityCurveTableConverterStation1:
+        | UpdateReactiveCapabilityCurveTable
+        | undefined;
+    updatePreviousReactiveCapabilityCurveTableConverterStation2:
+        | UpdateReactiveCapabilityCurveTable
+        | undefined;
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/vsc/modification/vsc-modification-from.tsx:41]:
```suggestion
    updatePreviousReactiveCapabilityCurveTableConverterStation1?:
        | UpdateReactiveCapabilityCurveTable
    updatePreviousReactiveCapabilityCurveTableConverterStation2?:
        | UpdateReactiveCapabilityCurveTable
```

File: src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx (hunk 9 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx:342]: handleSubmit is already a function provided by react-form-hook, it used like that in there documentation: https://react-hook-form.com/docs/useform/handlesubmit

File: src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx (hunk 6 above)
Comment (jamal-khey) [reply src/components/dialogs/network-modifications/vsc/creation/vsc-creation-form.tsx]: done

**Review by lilian-houdelet — APPROVED** [review]

**Review by YenguiSeddik — APPROVED** [review]

**Pull request merged**